setup.py

```diff
@@ -715,6 +715,12 @@ def detect_modules(self):
         # syslog daemon interface
         exts.append( Extension('syslog', ['syslogmodule.c']) )
 
+        # Fuzz tests.
+        exts.append( Extension(
+            '_fuzz',
```

Comment:
What prevents the _fuzz extension module from being installed and shipped with CPython?  It seems like something needed only at build + test time in order to prevent bitrot of code that is primarily used outside of our own test suite. Not something to be distributed.

Reply:
Nothing. I don't think it should be distributed with Python, but I don't know how to stop that while still keeping it built for tests. Any suggestions?

Reply:
I'm not sure the `_fuzz` module and it's corresponding test_fuzz.py add any real value.  I expect you could make `fuzzer.c` be compile-only (with no attempt to link it into the main binary) with a Makefile.pre.in modification which should be good enough.  At that point it could live in the `Programs/` directory next to with `_testembed.c` with which it seems more akin to instead of under `Modules/`

A test that calling the fuzz function with a trivial input in our own test suite not crashing is not so important as [the oss-fuzz project](https://github.com/google/oss-fuzz/blob/master/docs/ideal_integration.md#build-support) by its very nature is going to be doing that so we'd be notified. That we at least keep the code compiling is pretty good.

Reply:
I mean, we can do this. It isn't ideal / isn't what oss-fuzz would like us to do. See e.g. this doc: https://github.com/google/oss-fuzz/blob/master/docs/ideal_integration.md#regression-testing  AIUI the oss-fuzz project doesn't want to be in the game of reporting broken tests, only of reporting actual security issues.

(I'm already deviating fairly substantially from what's ideal, but I'd like to be as close as possible).

Reply:
Alright, talking to @ncoghlan at the dev. sprint this week, we've got other modules that are built and effectively "shipped" but are really used for testing/development only.  so this isn't the first, second, or even third such thing.  `_test_capi` is one of them.  `xxlimited` is another.  so i'm going to propose _xx_test_fuzz as the name to "Use all the naming idioms!" </meme>.  Python distribution creators are expected to realize that isn't needed in the main interpreter runtime package.

... let me rename things and look at this in whole again from that point of view.